extension/awsecshealthcheckextension/ecshealthcheckextension.go

```diff
@@ -0,0 +1,141 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecshealthcheckextension
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+	"go.uber.org/zap"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
+)
+
+type ECSHealthCheckExtension struct {
+	config   Config
+	logger   *zap.Logger
+	state    *healthcheck.HealthCheck
+	server   http.Server
+	stopCh   chan struct{}
+	exporter *ECSHealthCheckExporter
+}
+
+// initViews function could register the "failed to send" view and the customized exporter
+func (hc *ECSHealthCheckExtension) initViews() error {
+	if err := view.Register(&view.View{
+		Name:        "AOC/ECSHealthCheckExporterFailedToSendSpans",
+		Description: "number of errors in exporters over span",
+		Measure:     obsmetrics.ExporterFailedToSendSpans,
+		Aggregation: view.Count(),
+	}); err != nil {
+		log.Fatalf("Failed to register view: %v", err)
+		return err
+	}
```

Comment:
Is this the only code that needs to access the internal package?

Reply:
@bogdandrutu yeah, i think this is the only thing we need to get from internal

Reply:
Why not simply just adding to https://github.com/open-telemetry/opentelemetry-collector/blob/main/internal/obsreportconfig/obsmetrics/obs_exporter.go as a generic metric, then you don't need to do anything else.

Reply:
okay so you mean if i add our new view here in the internal package, i will be able to use it even in the contrib repo?

Reply:
Hey @bogdandrutu ! i checked and found that i can add our new view here https://github.com/open-telemetry/opentelemetry-collector/blob/main/internal/obsreportconfig/obsreportconfig.go#L104-L105 due to we have to use view.Count(), so the code will be:

```
errorNumberView := &view.View{
		Name:        "ExporterFailedNumber",
		Description: "number of errors in exporters over span",
		Measure:     obsmetrics.ExporterFailedToSendSpans,
		Aggregation: view.Count(),
	}
views = append(views, errorNumberView)
```
is it correct? if so, i will push a new pr for this

Reply:
Yes please

Reply:
hey @bogdandrutu ! found another issue that how can we get the new view in opentelemetry-collector-contrib if we follow this way? since all the obsmetrics embedded in the obsreport in the exporter helper from each exporter, i still cannot get the obsmetrics from opentelemetry-collector internal folder.

Reply:
hey @bogdandrutu ! can i ask another question how can we get the new view in opentelemetry-collector-contrib if we follow this way? since all the obsmetrics embedded in the obsreport in the exporter helper from each exporter, we still cannot get the obsmetrics from opentelemetry-collector internal folder. The only way i think could work is to add a new function or change the current function in the obsreport component to export the new metrics to our health check, is this way work to you?

Reply:
I am completely lost.

Reply:
Why is this not enough? https://github.com/open-telemetry/opentelemetry-collector/pull/3769?

Reply:
You don't access that view anywhere else

Reply:
hey @bogdandrutu! so we have the new view in obsmetrics in the main repo and where is this metric sent to? we need to collect this new view in our new health check extension. We cannot even get the name of our new view now.

Our original code was:
```
if err := view.Register(&view.View{
		Name:        "AOC/ECSHealthCheckExporterFailedToSendSpans",
		Description: "number of errors in exporters over span",
		Measure:     obsmetrics.ExporterFailedToSendSpans,
		Aggregation: view.Count(),
	}); err != nil {
		log.Fatalf("Failed to register view: %v", err)
		return err
	}

	hc.exporter = newECSHealthCheckExporter()
	view.RegisterExporter(hc.exporter)
	return nil
``` 

so originally we used our view to register our new exporter,  how can we do that in the new idea?

Reply:
You don't use your view, you use the `view` package to register your exporter. See the fact that on line 44 you don't keep a reference on the view instance. You need to rely on the name in the exporter to access the data for this specific view.

Reply:
hey @bogdandrutu ! so based on your words in the meeting, we still need to register the view and use this view register the exporter, right?

Reply:
got it, i will push 2 new pr for the changes in opentelemetry-collector and opentelemetry-collector-contrib repo, thanks!